### ISSUE TITLE
FactoryOf being able to resolve a dependency that needs to use parametersOf

### ISSUE THREAD

**Issue by lotdrops** [InsertKoinIO/koin#1325]
**Is your feature request related to a problem? Please describe.**
I often declare factories/singles/scopes where a parameter needs to be declared, not to construct such object, but to construct its dependencies.  
Example:
```
scoped { (navParam: Type ->
    ExampleViewModel(
        myParameter = get { parametersOf(navParam) }
    )
}
factory { (navParam: Type ->
    NewType(navParam)
}
```

**Describe the solution you'd like**
I'd like to be able to use `scopedOf(::ExampleViewModel)` (and the same with `factoryOf` and `singleOf`).

**Describe alternatives you've considered**
A possible way of simplifying this would be having a way to specify in the lambda how to get NewType from Type. Maybe something like a local factory would work:
```
scopedOf(::ExampleViewModel) { (navParam: Type) ->
  withFactory { get<NewType> { parametersOf(navParam) } }
}
```

**Target Koin project**
Koin 3.2

**Comment by arnaudgiuliani** [comment]
it's impacting graph resolution engine. I reschedule for later release

**Comment by arnaudgiuliani** [comment]
rescheduling for 3.5. Need to check again 👍

**Comment by arnaudgiuliani** [comment]
propagated parameters is available since 3.4.3. You can cascade params into sub-definitions. Please check and reopen if needed
